app/modules/stride/controllers/actions/GatekeeperAuthorisationActions.scala

```diff
@@ -66,7 +66,16 @@ trait GatekeeperAuthorisationActions {
         val retrieval = Retrievals.name and Retrievals.authorisedEnrolments
 
         authConnector.authorise(predicate, retrieval) map {
-          case Some(name) ~ authorisedEnrolments => Right(new LoggedInRequest(name.name, authorisedEnrolments, request))
+          case Some(name) ~ authorisedEnrolments => 
+            def applyRole(role: GatekeeperRole): Either[Result, LoggedInRequest[A]] =  Right(new LoggedInRequest(name.name, role, request))
+
+            ( authorisedEnrolments.getEnrolment(strideAuthConfig.adminRole).isDefined, authorisedEnrolments.getEnrolment(strideAuthConfig.superUserRole).isDefined, authorisedEnrolments.getEnrolment(strideAuthConfig.userRole).isDefined ) match {
```

Comment:
is there any significant cost (eg db/network call ) to calling authorisedEnrolments.getEnrolment? we always call it 3 times here even if the 'adminRole' value returns true making the other calls redundant

Reply:
Nope - enrolments is created with a set of enrolments and these are just looking for an entry in that set.
So no db or network overhead for each lookup.